### PR TITLE
Refresh balloon toolbar after change in the editor

### LINKED ISSUE #1342: Balloon Toolbar should be re-positioned after a change event
## Are you reporting a feature request or a bug?

Bug

## Provide detailed reproduction steps (if any)

1. Checkout d005e776b13cd7e6d0dcfa61b2cd9f5c8c129201.
2. Open http://ckeditor.test/ckeditor-dev/plugins/easyimage/samples/easyimage.html
2. Click image.
3. Change image to side one using balloon toolbar.

### Expected result

Balloon toolbar changes position to follow image widget.

### Actual result

Balloon toolbar stays in the same position, pointing to nowhere…

## Other details

* Browser: …
* OS: …
* CKEditor version: 4.8.0
* Installed CKEditor plugins: balloontoolbar

Note that there is proposed solution in bc37aeb5345353cd6e035e12698ccab891cda4c1 and 7dcba65224193bae5de991cd57022ce4d7e4070a, which should be a good starting point.

### PR DESCRIPTION
**Pull request by Comandeer**
## What is the purpose of this pull request?

Bug fix

## Does your PR contain necessary tests?

All patches which change the editor code must include tests. You can always read more
on [PR testing](https://docs.ckeditor.com/ckeditor4/docs/#!/guide/dev_contributing_code-section-tests),
[how to set the testing environment](https://docs.ckeditor.com/ckeditor4/docs/#!/guide/dev_tests) and
[how to create tests](https://docs.ckeditor.com/ckeditor4/docs/#!/guide/dev_tests-section-creating-your-own-test)
in the official CKEditor documentation.

### This PR contains

- [x] Unit tests
- [x] Manual tests

## What changes did you make?

Added listener for `change` to balloon toolbar position refresher. The drawback of the current solution is the fact that balloon toolbar without `undo` plugin won't refresh its position dynamically.

Closes #1342.

### CODE REVIEW

File: tests/plugins/balloontoolbar/refreshposition.js [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+		}
+	};
+
+	var parentFrame = window.frameElement,
+		originalHeight = parentFrame && parentFrame.style.height;
+
+	var tests = {
+		setUp: function() {
+			// In IE8 tests are run in very small window which breaks positioning assertions and tests fails (#1076).
+			if ( CKEDITOR.env.ie && CKEDITOR.env.version < 9 ) {
```
Comment (mlewand) [inline tests/plugins/balloontoolbar/refreshposition.js]: This part is done by `ignoreUnsupportedEnvironment` function that you call at the end of a file.

File: tests/plugins/balloontoolbar/refreshposition.js [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+				assert.ignore();
+			}
+			if ( parentFrame ) {
+				parentFrame.style.height = '900px';
```
Comment (mlewand) [inline tests/plugins/balloontoolbar/refreshposition.js]: Do we really need these tricks? I see it's a copy'n'paste from positioning.js, to get around #1076, however I cannot reproduce issue described in this ticket.

File: tests/plugins/balloontoolbar/refreshposition.js [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+			if ( parentFrame ) {
+				parentFrame.style.height = originalHeight;
+			}
+		},
+
+		// #1342
+		'test panel refresh position': function( editor, bot ) {
+
+			bot.setData( '<figure class="image"><img src="' + bender.basePath + '/_assets/lena.jpg"></figure>', function() {
+				var balloonToolbar = new CKEDITOR.ui.balloonToolbarView( editor, {
```
Comment (mlewand) [inline tests/plugins/balloontoolbar/refreshposition.js]: Wrong indentation after this var.

File: tests/plugins/balloontoolbar/manual/refreshposition.html [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+					editor.balloonToolbars.create( {
+						buttons: 'Link,Unlink',
+						widgets: [ 'image' ]
+					} );
+
+					editor.widgets.instances[ 0 ].on( 'focus', function( evt ) {
+						setTimeout( function() {
+							editor.execCommand( 'sideImage', { widget: evt.sender } );
+						}, 5000 );
```
Comment (mlewand) [inline tests/plugins/balloontoolbar/manual/refreshposition.html]: Can we make it 3 secs? 5 seems like forever :D

Comment (Comandeer) [reply tests/plugins/balloontoolbar/manual/refreshposition.html]: I've made it even 2 seconds ;)

**Comment by Comandeer** [comment]
I've moved unit test to `positioning.js` as it's connected with toolbar position.

**Comment by mlewand** [comment]
Rebased onto latest master branch.

**Pull request merged**